app/code/Magento/Catalog/Ui/DataProvider/Product/Form/Modifier/Eav.php

```diff
@@ -399,7 +400,11 @@ public function modifyData(array $data)
             foreach ($attributes as $attribute) {
                 if (null !== ($attributeValue = $this->setupAttributeData($attribute))) {
                     if ($attribute->getFrontendInput() === 'price' && is_scalar($attributeValue)) {
-                        $attributeValue = $this->formatPrice($attributeValue);
+                        if ($this->locator->getProduct()->getTypeId() !== ProductType::TYPE_BUNDLE
```

Comment:
By adding an additional complex condition here we increase the cyclomatic complexity of the method. This approach needs to be refactored (moved to a private method or so).